yb-voyager/src/tgtdb/attr_name_registry.go

```diff
@@ -0,0 +1,107 @@
+package tgtdb
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	"golang.org/x/exp/slices"
+)
+
+type AttributeNameRegistry struct {
+	dbType    string
+	tdb       TargetDB
+	tconf     *TargetConf
+	attrNames map[string][]string
+}
+
+func NewAttributeNameRegistry(tdb TargetDB, tconf *TargetConf) *AttributeNameRegistry {
+	return &AttributeNameRegistry{
+		dbType:    tconf.TargetDBType,
+		tdb:       tdb,
+		tconf:     tconf,
+		attrNames: make(map[string][]string),
+	}
+}
+
+func (reg *AttributeNameRegistry) QuoteIdentifier(schemaName, tableName, columnName string) string {
+	var err error
+	qualifiedTableName := fmt.Sprintf("%s.%s", schemaName, tableName)
+	targetColumns, ok := reg.attrNames[qualifiedTableName]
+	if !ok {
+		targetColumns, err = reg.tdb.GetListOfTableAttributes(schemaName, tableName)
+		if err != nil {
+			utils.ErrExit("get list of table attributes: %w", err)
+		}
+		reg.attrNames[qualifiedTableName] = targetColumns
+	}
+	c, err := reg.findBestMatchingColumnName(columnName, targetColumns)
+	if err != nil {
+		utils.ErrExit("find best matching column name for %q in table %s.%s: %w", columnName, schemaName, tableName, err)
+	}
+	return fmt.Sprintf("%q", c)
+}
+
+func (reg *AttributeNameRegistry) IfRequiredQuoteColumnNames(tableName string, columns []string) ([]string, error) {
```

Comment:
Let's rename this to `QuoteAttributeNames()` and reimplement this using calls to `QuoteAttributeName()` in a loop over `columns`.